### PR TITLE
Rewrite module with typescript

### PR DESCRIPTION
**Pull request by jeromeheissler**
Make new rollup config to export esm/cjs and iife module. Export module to manually register it with quill. Make custom exports to keep autoregister functionality. Upgrade to Quill 2

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,47 +3,69 @@
   "version": "4.1.0",
```
Comment (cavasinf) [inline package.json:3]: Would be nice in `5.0.0` if it switch to Quill 2.0

**Comment by csculley** [comment]
This could close #362

**Comment by cavasinf** [comment]
And maybe close #344

**Comment by csculley** [comment]
This will also close #348

**Comment by MentalGear** [comment]
Thanks your work on this and mentions in general is much appreciated!
Could you elaborate on how you'd like testing to be done, writing unit tests or manual testing?

**Comment by artknight** [comment]
Hello Fellas,
Apologies for the late reply, but I just glazed over the [Artknight changes](https://github.com/quill-mention/quill-mention/pull/375) and seems the PR is missing a bunch of changes. (https://github.com/artknight/quill-emoji/commit/bfff17cbb82c4ce2bb974eb657013c7133ae85fa). 

That being said, if everyone wants me to merge all **my original changes** into this repo, I can do that. Please let me know your thoughts

**Comment by dan-bartlettroofs** [comment]
> Hello Fellas, Apologies for the late reply, but I just glazed over the [Artknight changes](https://github.com/quill-mention/quill-mention/pull/375) and seems the PR is missing a bunch of changes as compared to my cloned repo. ([artknight/quill-emoji@bfff17c](https://github.com/artknight/quill-emoji/commit/bfff17cbb82c4ce2bb974eb657013c7133ae85fa)).
> 
> That being said, if everyone wants me to merge all **my original changes** into this repo, I can do that. Please let me know your thoughts

Please do!

**Comment by artknight** [comment]
Here is the [PR](https://github.com/quill-mention/quill-mention/pull/376). For sanity, I just need someone to review the changes real quick...

**Comment by MadSpindel** [comment]
@jeromeheissler Maybe we should first merge something like PR #375, then investigate if we can get your other fixes in separate PRs?

**Comment by jeromeheissler** [comment]
@MadSpindel I've rebased my code to the latest commit on master

**Review by MadSpindel — APPROVED** [review]
LGTM! Excellent job @jeromeheissler!

**Pull request merged**